ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/requests/RequestProcessor.java

```diff
@@ -15,14 +15,22 @@
 package org.hyperledger.besu.ethereum.mainnet.requests;
 
 import org.hyperledger.besu.ethereum.core.MutableWorldState;
+import org.hyperledger.besu.ethereum.core.ProcessableBlockHeader;
 import org.hyperledger.besu.ethereum.core.Request;
 import org.hyperledger.besu.ethereum.core.TransactionReceipt;
+import org.hyperledger.besu.ethereum.mainnet.ProtocolSpec;
+import org.hyperledger.besu.evm.operation.BlockHashOperation.BlockHashLookup;
+import org.hyperledger.besu.evm.tracing.OperationTracer;
 
 import java.util.List;
 import java.util.Optional;
 
 public interface RequestProcessor {
   Optional<List<? extends Request>> process(
+      final ProcessableBlockHeader blockHeader,
       final MutableWorldState mutableWorldState,
-      final List<TransactionReceipt> transactionReceipts);
+      final ProtocolSpec protocolSpec,
+      final List<TransactionReceipt> transactionReceipts,
+      final BlockHashLookup blockHashLookup,
+      final OperationTracer operationTrace);
 }
```

Comment:
One thing I do not like when we have many parameters in a method like this is that every time we need to add/remove anything, it changes in a bunch of places. I'd consider encapsulating these parameters into something like `RequestProcessorContext` or `RequestProcessorParameters`. The same probably applies to all other process(a, b, c, d, e, f) methods.